### DISCUSSION TITLE
Script to cycle through the workspaces.

### DISCUSSION THREAD

**Discussion by nnuel** [Manjaro-Sway/manjaro-sway#228] [Ideas]
Not sure how to tag this issue as an idea, but I found a script on Reddit and made it work with manjaro-sway:

```
#!/bin/sh    
    
CUR_WORKSPACE=$(swaymsg -t get_outputs | jq -r '.[] | select(.focused)' | jq -r '.current_workspace')    
    
if [ $1 == "prev" ]; then    
    TO_WORKSPACE=$(expr ${CUR_WORKSPACE:0:1} - 1)    
elif [ $1 == "next" ]; then    
    TO_WORKSPACE=$(expr ${CUR_WORKSPACE:0:1} + 1)    
fi    
    
if [ $TO_WORKSPACE -eq 10 ]; then    
    TO_WORKSPACE=0
elif [ $TO_WORKSPACE -eq -1 ]; then    
    TO_WORKSPACE=9    
fi    
    
swaymsg workspace number $TO_WORKSPACE  
```

I have set the Workspace #10 to workspace 0, and there must be ways to change this script to work with the default 1-10, but I like the single digit set-up, so I will test that out for now.

The scripts gets called upon by these bindsyms:

```
# To cycle through all workspaces:
bindsym $mod+tab exec "~/scriptname.sh next"
bindsym $mod+shift+tab exec "~/scriptname.sh prev"

# To cycle through active workspaces:
bindsym mod1+tab workspace next
bindsym mod1+shift+tab workspace prev
```

Lastly, how can you overwrite the default keybinds. These don't seem to work for me:

```
bindsym $mod+d workspace ""
bindsym $mod+s exec $menu
```

**Comment by Mathew-D** [comment]
You have to unbindsym before you can change a binding.
`$unbindsym $mod+s`

**Comment by boredland** [comment]
afaik this overlaps a lot with swayr, which I have proposed an an off the shelf solutions for this [here](https://github.com/Manjaro-Sway/manjaro-sway/discussions/184) but personally never felt the need for, but am certainly open for PRs into that direction or other ideas.

**Comment by nnuel** [comment]
Yes, on the one hand it does. At first sight, swayr looks great and powerful, but it needs some setting up to do, and this is what would bring alt+tab behaviour as a recent convert to linux from windows, so this was something I missed. I am also quite sure that I don't use workspaces as much as they are intended to be used, but I am learning. For the time being it at least helps me navigate quickly, for when I toggle the waybar, and can't see on which workspace there are my various windows openend. 

I am still looking for a good solution to open the next emptyworkspace. For example, 1, 2 and 4 are used, and I want it to open in a new empty workspace - the first one available. I will have a think, because I can't seem to find that in swayr and it would probably involve something with swaymsg. It would need to make an array of the various workspaces, and compare it to an array of 1-10 and then grab the lowest one. It would make sense to bind this to 

```
$mod + `
```
And a version with 
```
$mod + shift + `
``` 
could be to push a window to the next open one. 

Not sure how to escape the ` in markdown with an in-line quote, so this should work regardless.

**Comment by boredland** [comment]
I think this is mostly covered already now. I use Alt+Tab a lot.
